internal/orchestrator/failure_recoverer.go

```diff
@@ -59,10 +61,17 @@ func (fr *FailureRecoverer) Start() {
 				blocksToTrigger = append(blocksToTrigger, blockFailure.BlockNumber)
 			}
 
+			// Trigger worker for recovery
 			log.Debug().Msgf("Triggering Failure Recoverer for blocks: %v", blocksToTrigger)
 			worker := worker.NewWorker(fr.rpc)
 			results := worker.Run(blocksToTrigger)
+			p := NewPoller(fr.rpc, fr.storage)
+			p.handleWorkerResults(results)
```

Comment:
This is hacky - `fr.handleWorkerResults` should be reviewed and implemented to store the staging data.
Do you want to do it here? I can also create a new PR on top of this and keep this as it is now